src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -281,10 +281,50 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     return this._loadPromise;
   }
 
+  /**
+   * handle decode error - reload the video and seek to last currentTime
+   * @returns {void}
+   */
+  handleDecodeError(): void {
+    NativeAdapter._logger.debug(`handleDecodeError currentTime <${this._videoElement.currentTime}>`);
+    this._playerTracks = this._getParsedTracks();
+    const prevCurrTime = this._videoElement.currentTime;
+    const prevActiveAudioTrack = this._getActivePKAudioTrack();
+    const prevActiveTextTrack = this._getActivePKTextTrack();
+
+    this._eventManager.removeAll();
+    this._loadPromise = null;
```

Comment:
don't you need to reset more members?
```javascript
this._waitingEventTriggered = false;
this._liveEdge = 0;
this._clearHeartbeatTimeout();
if (this._liveDurationChangeInterval) {
  clearInterval(this._liveDurationChangeInterval);
  this._liveDurationChangeInterval = null;
}
```

What about the DRM handler? if you reset a DRM playback does it recover correctly? need to check this